compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.td

```diff
@@ -250,4 +251,89 @@ def IREEGPU_GlobalLoadDMAOp : Op<IREEGPU_Dialect, "global_load_dma", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// CoalescedGatherDMAOp
+//===----------------------------------------------------------------------===//
+
+def IREEGPU_CoalescedGatherDMAOp : Op<IREEGPU_Dialect, "coalesced_gather_dma", [
+    DeclareOpInterfaceMethods<DestinationStyleOpInterface>]> {
+  let summary = "Coalesced gather DMA operation for efficient GPU memory access";
+  let description = [{
+    Performs a coalesced gather operation that reads elements from a source tensor
+    based on indices and writes them to a destination tensor. This operation is
+    designed to maximize memory bandwidth utilization on GPU hardware through
+    coalesced memory access patterns. Conceptually:
+
+    for %dim_indices in dimension($init):
```

Comment:
This pseudocode is worded weird and it's unclear what's going on.

Also, this leaves things like the transfer size and the fact that things are somewhat tiled to subgroups unspecified